From f51b844c8291dc69099bc5620ea4780b0e945df6 Mon Sep 17 00:00:00 2001
From: Mike Achtelik <mike.achtelik@gmail.com>
Date: Thu, 25 Mar 2021 16:28:58 +0100
Subject: Set android nfc transceive timeout

As it stands, each device uses its own timeout for the transceive command.
While most devices seem to use 618 ms, there are some devices which choose
a timeout of something like 309 ms.
So lets set a common timeout, so the behavior stays consistent between
devices. Use 2000 ms as that should be enough for most applications.

Change-Id: I4fae8f284b9df6c78f7117e966dd9fed2f48fd9d
Reviewed-by: Alex Blasche <alexander.blasche@qt.io>
(cherry picked from commit d15693d4ea2e06216f55facd504273c990b3814b)
---
 src/nfc/qnearfieldtarget_android.cpp | 11 +++++++++++
 src/nfc/qnearfieldtarget_android_p.h |  1 +
 2 files changed, 12 insertions(+)

diff --git x/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp y/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
index e42671eb..edca3753 100644
--- x/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
+++ y/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
@@ -524,10 +524,21 @@ bool NearFieldTarget::connect()
     if (connected)
         return true;

+    setCommandTimeout(2000);
     m_tagTech.callMethod<void>("connect");
     return !catchJavaExceptions();
 }

+bool NearFieldTarget::setCommandTimeout(int timeout)
+{
+    if (!m_tagTech.isValid())
+        return false;
+
+    m_tagTech.callMethod<void>("setTimeout", "(I)V", timeout);
+
+    return !catchJavaExceptions();
+}
+
 QByteArray NearFieldTarget::jbyteArrayToQByteArray(const jbyteArray &byteArray) const
 {
     QAndroidJniEnvironment env;
diff --git x/qtconnectivity/src/nfc/qnearfieldtarget_android_p.h y/qtconnectivity/src/nfc/qnearfieldtarget_android_p.h
index 0063e9a5..27a85029 100644
--- x/qtconnectivity/src/nfc/qnearfieldtarget_android_p.h
+++ y/qtconnectivity/src/nfc/qnearfieldtarget_android_p.h
@@ -104,6 +104,7 @@ protected:
     QAndroidJniObject getTagTechnology(const QString &tech) const;
     bool setTagTechnology(const QStringList &techList);
     bool connect();
+    bool setCommandTimeout(int timeout);
     QByteArray jbyteArrayToQByteArray(const jbyteArray &byteArray) const;
     bool catchJavaExceptions(bool verbose = true) const;

--
2.33.0

